src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.cs

```diff
@@ -135,7 +139,10 @@ private HttpClient CreateHttpClient()
                 }
             }
 
-            var httpClient = httpMessageHandler == null ? new HttpClient() : new HttpClient(httpMessageHandler);
+            // Wrap message handler in a logging handler last to ensure it is always present
+            httpMessageHandler = new LoggingHttpMessageHandler(httpMessageHandler, _loggerFactory);
```

Comment:
Might be worth checking if `_loggerFactory` is a `NullLoggerFactory`. Not necessary for this PR though